src/prefect/indicators/infrastructure/i1.py

```diff
@@ -0,0 +1,203 @@
+        ids = targets["id"].apply(UUID)  # type: ignore[arg-type]
+        chunks = (
+            np.array_split(ids, int(len(ids) / chunk_size))
+            if len(ids) > chunk_size
+            else [ids.to_numpy()]
+        )
+        futures = [
+            get_points_of_charge_for_targets.submit(connection, level, chunk)  # type: ignore[call-overload]
+            for chunk in chunks
+        ]
+        wait(futures)
+
+    # Concatenate results
+    results = pd.concat([future.result() for future in futures], ignore_index=True)
+
+    # Serialize indicators
+    merged = targets.merge(results, how="left", left_on="id", right_on="level_id")
+    merged["num_poc"] = merged["num_poc"].fillna(0)
+
+    # Build result DataFrame
+    indicators = pd.DataFrame()
+    indicators["target"] = merged["code"]
+    indicators["value"] = merged["num_poc"]
+    indicators["code"] = "i1"
+    indicators["level"] = level
+    indicators["period"] = period
+    indicators["timestamp"] = at
+    indicators[["category", "extras"]] = None
+
+    return indicators
+
+
+@flow(
+    task_runner=ThreadPoolTaskRunner(max_workers=settings.THREAD_POOL_MAX_WORKERS),
+    flow_run_name="i1-{period.value}-00-{at:%y-%m-%d}",
+    log_prints=True,
+)
+def i1_national(period: IndicatorPeriod, at: datetime) -> pd.DataFrame:
+    """Calculate i1 at the national level."""
+    engine = get_database_engine()
+    with engine.connect() as connection:
+        result = connection.execute(text("SELECT COUNT(*) FROM PointDeCharge"))
+        count = result.one()[0]
+
+    return pd.DataFrame.from_records(
+        [
+            Indicator(
+                code="i1",
+                level=Level.NATIONAL,
+                period=period,
+                value=count,
+                timestamp=at,
+            ).model_dump(),
+        ]
+    )
+
+
+@flow(
+    task_runner=ThreadPoolTaskRunner(max_workers=settings.THREAD_POOL_MAX_WORKERS),
+    flow_run_name="meta-i1-{period.value}",
+)
+def calculate(
+    period: IndicatorPeriod, create_artifact: bool = False, chunk_size: int = 1000
+) -> List[Indicator]:
+    """Run all i1 subflows."""
+    now = pd.Timestamp.now()
+    national = i1_national(period, now)
+    regions = i1_for_level(Level.REGION, period, now, chunk_size=chunk_size)
+    departments = i1_for_level(Level.DEPARTMENT, period, now, chunk_size=chunk_size)
+    epcis = i1_for_level(Level.EPCI, period, now, chunk_size=chunk_size)
+    cities = i1_for_level(Level.CITY, period, now, chunk_size=chunk_size)
+
+    indicators = pd.concat(
+        [national, regions, departments, epcis, cities], ignore_index=True
+    )
+    if create_artifact:
+        create_markdown_artifact(
+            key=runtime.flow_run.name,
+            markdown=indicators.to_markdown(),
+            description=f"i1 report at {now} (period: {period.value})",
+        )
+
+    return [Indicator(**record) for record in indicators.to_dict(orient="records")]  # type: ignore[misc]
```

Comment:
- in the calculate function, would it be interesting to keep only the lowest level (and to calculate the other levels if necessary) ?

Reply:
Do you mean adding more options to the `calculate` function so that we can toggle computation?

Reply:
As discussed IRL, this is something that will evolve in the coming weeks.